ambry-tools/src/main/java/com.github.ambry/store/DumpIndexTool.java

```diff
@@ -42,481 +49,358 @@
-      String blobIdsStr = verifiableProperties.getString("blob.id.list", "");
-      List<String> blobIds = blobIdsStr.isEmpty() ? Collections.EMPTY_LIST : Arrays.asList(blobIdsStr.split(","));
-      boolean activeBlobsOnly = verifiableProperties.getBoolean("active.blobs.only", false);
-      boolean silent = verifiableProperties.getBoolean("silent", true);
-      logger.info("Type of Operation - {}", typeOfOperation);
-      if (blobIds.size() > 0) {
-        logger.info("Blobs to look out for - {}", blobIds);
-      }
-
-      Time time = SystemTime.getInstance();
+        clusterMapConfig, config.hardwareLayoutFilePath, config.partitionLayoutFilePath)).getClusterMap()) {
+      StoreConfig storeConfig = new StoreConfig(verifiableProperties);
+      // this tool supports only blob IDs. It can become generic if StoreKeyFactory provides a deserFromString method.
+      BlobIdFactory blobIdFactory = new BlobIdFactory(clusterMap);
       StoreToolsMetrics metrics = new StoreToolsMetrics(clusterMap.getMetricRegistry());
-      reporter = JmxReporter.forRegistry(clusterMap.getMetricRegistry()).build();
-      reporter.start();
-      DumpIndexTool dumpIndexTool = new DumpIndexTool(clusterMap, silent, time, metrics);
-      switch (typeOfOperation) {
-        case "DumpIndexFile":
-          String fileToRead = verifiableProperties.getString("file.to.read");
-          logger.info("File to read " + fileToRead);
-          if (activeBlobsOnly) {
-            dumpIndexTool.dumpActiveBlobsFromIndex(new File(fileToRead), blobIds, time.milliseconds());
-          } else {
-            dumpIndexTool.dumpIndex(new File(fileToRead), null, null, blobIds, null, new IndexStats(), false,
-                time.milliseconds());
-          }
+      StoreMetrics storeMetrics = new StoreMetrics("DumpIndexTool", clusterMap.getMetricRegistry(),
+          new AggregatedStoreMetrics(clusterMap.getMetricRegistry()));
+      Time time = SystemTime.getInstance();
+      DumpIndexTool dumpIndexTool = new DumpIndexTool(blobIdFactory, storeConfig, time, metrics, storeMetrics);
+      Set<StoreKey> filterKeySet = new HashSet<>();
+      for (String key : config.filterSet) {
+        filterKeySet.add(new BlobId(key, clusterMap));
+      }
+      switch (config.typeOfOperation) {
+        case DumpIndex:
+          dumpIndex(dumpIndexTool, config.pathOfInput, filterKeySet);
+          break;
+        case DumpIndexSegment:
+          dumpIndexSegment(dumpIndexTool, config.pathOfInput, filterKeySet);
           break;
-        case "DumpIndexesForReplica":
-          String replicaRootDirecotry = verifiableProperties.getString("replica.root.directory");
-          long indexEntriesPerSec = verifiableProperties.getLong("index.entries.per.sec", 1000);
-          if (activeBlobsOnly) {
-            dumpIndexTool.dumpActiveBlobsForReplica(replicaRootDirecotry, blobIds, time.milliseconds());
+        case VerifyIndex:
```

Comment:
have you documented somewhere, what does VerifyIndex mean

Reply:
It's a part of the class documentation but I have also added it to the enum now.